src/backend/glutin/mod.rs

```diff
@@ -63,53 +63,56 @@ impl Display {
     /// by the implementation.
     pub fn new(
         wb: glutin::WindowBuilder,
-        cb: glutin::ContextBuilder,
+        cb: glutin::ContextBuilder<NotCurrent>,
         events_loop: &glutin::EventsLoop,
     ) -> Result<Self, DisplayCreationError>
     {
-        let gl_window = glutin::WindowedContext::new_windowed(wb, cb, events_loop)?;
+        let gl_window = cb.build_windowed(wb, events_loop)?;
+        let gl_window = unsafe {
+            gl_window.treat_as_current()
+        };
         Self::from_gl_window(gl_window).map_err(From::from)
     }
 
     /// Create a new glium `Display`.
     ///
     /// Performs a compatibility check to make sure that all core elements of glium are supported
     /// by the implementation.
-    pub fn from_gl_window(gl_window: glutin::WindowedContext) -> Result<Self, IncompatibleOpenGl> {
+    pub fn from_gl_window(gl_window: glutin::WindowedContext<Pc>) -> Result<Self, IncompatibleOpenGl> {
         Self::with_debug(gl_window, Default::default())
     }
 
     /// Create a new glium `Display`.
     ///
     /// This function does the same as `build_glium`, except that the resulting context
     /// will assume that the current OpenGL context will never change.
-    pub unsafe fn unchecked(gl_window: glutin::WindowedContext) -> Result<Self, IncompatibleOpenGl> {
+    pub unsafe fn unchecked(gl_window: glutin::WindowedContext<Pc>) -> Result<Self, IncompatibleOpenGl> {
         Self::unchecked_with_debug(gl_window, Default::default())
     }
 
     /// The same as the `new` constructor, but allows for specifying debug callback behaviour.
-    pub fn with_debug(gl_window: glutin::WindowedContext, debug: debug::DebugCallbackBehavior)
+    pub fn with_debug(gl_window: glutin::WindowedContext<Pc>, debug: debug::DebugCallbackBehavior)
         -> Result<Self, IncompatibleOpenGl>
     {
         Self::new_inner(gl_window, debug, true)
     }
 
     /// The same as the `unchecked` constructor, but allows for specifying debug callback behaviour.
     pub unsafe fn unchecked_with_debug(
-        gl_window: glutin::WindowedContext,
+        gl_window: glutin::WindowedContext<Pc>,
         debug: debug::DebugCallbackBehavior,
     ) -> Result<Self, IncompatibleOpenGl>
     {
         Self::new_inner(gl_window, debug, false)
     }
 
     fn new_inner(
-        gl_window: glutin::WindowedContext,
+        gl_window: glutin::WindowedContext<Pc>,
```

Comment:
Context is assumed to be current. Fine and all, but what if they want to pass a not current one. This forces users to either pretend it is current with `treat_as_current` or actually make it current, with `make_current`, which might not be something they want.

This is not wrong per se, indeed this code would be functional.